src/aspire/denoising/denoised_src.py

```diff
@@ -1,31 +1,29 @@
 import logging
 
-import numpy as np
-
-from aspire.image import Image
+from aspire.denoising import Denoiser
 from aspire.source import ImageSource
 
 logger = logging.getLogger(__name__)
 
 
 class DenoisedImageSource(ImageSource):
     """
-    Define a derived ImageSource class to perform operations for denoised 2D images
+    `ImageSource` class serving denoised 2D images.
     """
 
-    def __init__(self, src, denoiser, batch_size=512):
+    def __init__(self, src, denoiser):
```

Comment:
Maybe. This didn't come up when you and Junchao were designing these, so I assumed it was how you wanted them already.  Wasn't really part of this issue, but I can change it while I'm in here.

Reply:
FWIW, this will block having any sensible default (would require a src) down the road.

Reply:
Looking at the change. This also breaks the pattern of an  FooSource being something that takes a Source and returns a Source after Foo. Not sure I like that.

Maybe would make more sense for the Foo subclasses that require the identical sources to just advertise that and the FooSource can confirm/reject whether the sources match when they need to. LMK thanks.

Reply:
> Looking at the change. This also breaks the pattern of an FooSource being something that takes a Source and returns a Source after Foo. Not sure I like that.

Right. I think it's valuable to keep this pattern if possible. There's two ways of going about this:

1. Consider the train–test perspective as described above. In other words, `denoiser` is defined (trained) using a certain source, but can be applied to any other source. This would require calculating the basis coefficients of the new images and applying the CWF to it. Possible, but unclear if this would be possible for other possible denoisers.

2. Have `denoiser` simple be a repository for parameters of a denoising algorithm, and then apply if to the `src` given here.

Looking at `OrientedSource`, I see that it has a similar issue (the source you supply to `OrientedSource` may have nothing to do with the one supplied to `CLOrient3D`). To your point, this PR is neither the time or place to deal with this, so closing for now.